### ISSUE TITLE
[DomCrawler] Ignore empty node exceptions

### ISSUE THREAD

**Issue by amer-flix** [symfony/symfony#28313]
**Description**  
When I try to access a node which does not exist in an XML / HTML, I get an exception:
**_The current node list is empty._**

Current Solution:
I have to add try and catch for each statement or I have to check the ->count() for each statement.

Is it possible to ignore this error so that if it doesn't exist, it just keeps it empty / null?

Thanks

**Example**  

// the following code throws an exception
$this->crawler->filter('marketingProductName')->text();

**Comment by nicolas-grekas** [comment]
Would you like to submit a PR going in this direction?

**Comment by amer-flix** [comment]
@nicolas-grekas 

I have submitted PR from my other account (@rajaamerkhan) 

I am returning null for all three methods (text, html and attr). This will be very convenient and help with avoiding a lot of repeated checks.

https://github.com/symfony/symfony/pull/28614

Please check.

Thanks

**Comment by nicolas-grekas** [comment]
Thanks, but actually there is already a discussion happening https://github.com/symfony/symfony/issues/28581!

**Comment by venusdemilo** [comment]
The Symfony 4.3 documentation now suggests the use of `text ('Default text content') `to avoid the exception. But that does not seem to work. The exception is always raised.

**Comment by venusdemilo** [comment]
Here is the patch that was not published in Symfony 4.3 :
```
 public function text()
    public function text(/* $default = null */)
    {
        if (!$this->nodes) {
            if (0 < \func_num_args()) {
                return \func_get_arg(0);
            }
            throw new \InvalidArgumentException('The current node list is empty.');
        }
```

**Comment by xabbuh** [comment]
@venusdemilo Please open a new issue if you think that you have found a bug. And please also add some instructions on how to reproduce it.
